Add WorldPlaceable component #82

mserier opened the pull request:
Add worldplaceable component to misslelauncher #82

mserier committed:
Add WorldPlaceable component #82

, add worldplaceable component to misslelauncher #82

mserier committed:
Merge branch 'develop' into feature/#82-tower-invalid-placement

mserier committed:
Change bitmask to use our own enum #82

mserier committed:
Merge branch 'develop' into feature/#82-tower-invalid-placement

mserier committed:
Fix that towers still could be placed in towers #82

leroykorterink commented on src/Assets/LightSkyDefense/Scripts/Common/Behaviours/WorldPlaceable.cs:
```diff
@@ -0,0 +1,64 @@
+        if (ObjectCollider == null)
+            return;
+
+        if (Physics.CheckBox(ObjectCollider.transform.position, ObjectCollider.size / 2f, transform.rotation, (int)_layerMask))
```
We can invert this if to avoid the nesting in else.

mserier replied:
Done but in new commit we still have to check if we transitioned from valid to invalid and vise-versa

leroykorterink commented on src/Assets/LightSkyDefense/Scripts/Common/Behaviours/WorldPlaceable.cs:
```diff
@@ -0,0 +1,64 @@
+
+        if (Physics.CheckBox(ObjectCollider.transform.position, ObjectCollider.size / 2f, transform.rotation, (int)_layerMask))
+        {
+            if (_originalPrefabMat == null)
```
Can't we create and save the materials in the start/awake method?

mserier replied:
Done

leroykorterink commented on src/Assets/LightSkyDefense/Scripts/Common/Behaviours/WorldPlaceable.cs:
```diff
@@ -0,0 +1,64 @@
+
+    void Awake()
+    {
+        if (!isVisual)
```
This if should be be inverted

mserier replied:
Done

mserier committed:
Store greymaterial and reuse it #82

Changed tower to sphere collider instead of boxcollider

Fix MissleLauncher scale (adjusted import settings)

mserier committed:
Merge branch 'develop' into feature/#82-tower-invalid-placement

hassanAlAwwadi committed:
Fix dial offset #82

Fix missle resize bug

hassanAlAwwadi committed:
Merge branch 'develop' into feature/#82-tower-invalid-placement

hassanAlAwwadi committed:
Merge develop into #82

hassanAlAwwadi committed:
Added second dial option #82

hassanAlAwwadi committed:
Reverted vrmanifest #82

mserier committed:
Refactored if statements in WorldPlaceable #82

mserier committed:
Half hour unnecessary refactor but slightly nicer #82
